runtime/lua/vim/snippet.lua

```diff
@@ -477,7 +477,9 @@ end
 function M.expand(input)
   local snippet = G.parse(input)
   local snippet_text = {}
-  local base_indent = vim.api.nvim_get_current_line():match('^%s*') or ''
+  local curbuf = vim.api.nvim_get_current_buf()
```

Comment:
Instead of adding a new variable could you move the declaration of `bufnr` here?

https://github.com/neovim/neovim/blob/f330eaaeb35557521b608c41fd95a2bb2a76972b/runtime/lua/vim/snippet.lua#L586